src/httpserver.rs

```diff
@@ -97,24 +97,23 @@ async fn handle(req: Request<Body>, snare: Arc<Snare>) -> Result<Response<Body>,
         return Ok(res);
     }
 
-    // We now check that there is a per-repo program for this repository and that we haven't been
-    // tricked into searching for a file outside of the repos dir.
+    // We now check that there is a per-repo program for this repository.
     let mut p = PathBuf::new();
     p.push(&conf.github.reposdir);
+    // The calls to github_valid_username and github_valid_reponame above guarantee that `owner`
+    // and `repo` are safe to put in pathnames.
```

Comment:
Would it also be worthwhile to escape user and repo names beginning with `~` as this expands to user homes. What about `$` in case the user can expand shell variables.

A possibly valid counter-example might be: we never (knowingly) pass these characters to a shell(?)

Reply:
`~` and `$` aren't valid characters, so they'll be rejected. Fortunately (and probably not coincidentally), GitHub's rules on this are onerous enough that nothing nasty can leak through.

Reply:
[The tests prove that these are rejected BTW: notice the `for i in 0..255` bits.]

Reply:
Ah ok, good!